pkg/reconciler/tenancy/clusterworkspaceshard/clusterworkspaceshard_controller.go

```diff
@@ -128,33 +128,36 @@ func (c *Controller) processNextWorkItem(ctx context.Context) bool {
 	// other workers.
 	defer c.queue.Done(key)
 
-	if err := c.process(ctx, key); err != nil {
+	if requeue, err := c.process(ctx, key); err != nil {
 		runtime.HandleError(fmt.Errorf("%q controller failed to sync %q, err: %w", ControllerName, key, err))
 		c.queue.AddRateLimited(key)
 		return true
+	} else if requeue {
+		// only requeue if we didn't error, but we still want to requeue
```

Comment:
could you describe applicable cases?

Reply:
this is a bit controversial. There was some logic for dealing with status modifications except that the reconcile loop was empty. I still kept the mechanism and the opportunity to flesh it out. I took the approach of doing 1 modification at once, meaning that it requeues each time a modification is done to give a chance to other chunks to do their processing independently. If no more modification is done no requeue.